File: 프로그래머스/연습문제/12933_정수_내림차순으로_배치하기.java
// https://school.programmers.co.kr/learn/courses/30/lessons/12933

// Todo: toCharArray(), StringBuilder(), reverse()

import java.util.Arrays;

class Solution {
    public long solution(long n) {
        char[] charArray = Long.toString(n).toCharArray();

        Arrays.sort(charArray);
        StringBuilder sortedStr = new StringBuilder(new String(charArray)).reverse();

        return Long.parseLong(sortedStr.toString());
    }
}

File: 프로그래머스/연습문제/12933_정수_내림차순으로_배치하기.cpp
// https://school.programmers.co.kr/learn/courses/30/lessons/12933

#include <string>
#include <vector>
#include <algorithm>

using namespace std;

long long solution(long long n) {
    string str = to_string(n);

    sort(str.begin(), str.end(), greater<>());

    return stoll(str);
}